src/main/java/org/apache/sysds/runtime/controlprogram/federated/monitoring/services/StatsService.java

```diff
@@ -45,13 +50,20 @@ public static BaseEntityModel getWorkerStatistics(Long id, String address) {
 					aggFedStats.aggregate((FederatedStatistics.FedStatsCollection)tmp[0]);
 
 				parsedStats = new StatsEntityModel(
-					id, aggFedStats.cpuUsage, aggFedStats.memoryUsage,
-					aggFedStats.heavyHitters, aggFedStats.coordinatorsTrafficBytes);
+						id,
+						new Timestamp(System.currentTimeMillis()),
+						aggFedStats.cpuUsage,
+						aggFedStats.memoryUsage,
+						aggFedStats.jitCompileTime,
+						aggFedStats.heavyHitters,
+						aggFedStats.coordinatorsTrafficBytes,
+						aggFedStats.requestTypeCount);
 			}
-		} catch(DMLRuntimeException dre) {
+		} catch (DMLRuntimeException dre) {
 			// silently ignore -> caused by offline federated workers
+			log.error("Worker offline: " + dre.getMessage());
 		} catch (Exception e) {
-			throw new RuntimeException(e);
+			log.error("Error: " + e.getMessage());
```

Comment:
this is super dangerous. 
Please verify that this is the actual behavior you want, 
can you limit the exception caught to something else?
And if it really is the functionality you want add a comment saying that we do not handle the exception but just log it, and that it is fine.

Reply:
Completely agree, for now, I left it as it is with additional comments, the problem was that there is a separate thread running for worker stats gathering and if the method fails, the thread stops, and no further data is gathered from any of the workers, the way to fix it is to restart the whole app. With this, the app can continue functioning without the restart.